src/Illuminate/Validation/Concerns/ValidatesAttributes.php

```diff
@@ -253,9 +253,15 @@ protected function compareDates($attribute, $value, $parameters, $operator)
         }
 
         if (is_null($date = $this->getDateTimestamp($parameters[0]))) {
+            $dateAttributes = explode(' ', $parameters[0]);
             $comparedValue = $this->getValue($parameters[0]);
 
-            if (! is_string($comparedValue) && ! is_numeric($comparedValue) && ! $comparedValue instanceof DateTimeInterface) {
+            if (count($dateAttributes) > 1) {
```

Comment:
Just explaining why `count() > 1` is enough here. We have the following options:

### User validates using `after:date_field`
This is the most straight forward case. The `$dateAttributes` array contains only one item and we don't execute the if block.

```
["date_field"]
```

### User validates `after:date_field +1day`

Now, `$dateAttributes` contains this array:
```
["date_field", "+1day"]
```

### User validates with `after:date_field +1 day`
In this scenario, `$dateAttributes` will be
```
["date_field", "+1", "day"]
```

### User validates with `after:date_field + 1 day`
Here, `$dateAttributes` contains
```
["date_field", "+", "1", "day"]
```